### PR TITLE
MAT-4598 version ID

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR Template

Jira Ticket: [MAT-4598](https://jira.cms.gov/browse/MAT-4598)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e. Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/models/measure/Measure.java [hunk 1]
```diff
@@ -98,6 +98,11 @@ public class Measure {
   private String model;
 
   private MeasureMetaData measureMetaData = new MeasureMetaData();
+  
+  @NotBlank(
+      groups = {ValidationOrder1.class},
+      message = "Version ID is required.")
+  private String versionId;
```
[the remark this answers is not in the document]
Comment (adongare) [reply src/main/java/gov/cms/madie/models/measure/Measure.java:105]: I'm asking this because if we plan to generate this on the server side, create measure call might get blocked because of this validation

[the remark this answers is not in the document]
Comment (adongare) [reply src/main/java/gov/cms/madie/models/measure/Measure.java:105]: also not sure if this is part of the metadata or measure model itself

[the remark this answers is not in the document]
Comment (sb-cecilialiu) [reply src/main/java/gov/cms/madie/models/measure/Measure.java:105]: @adongare Thanks a lot Ashok, those are good questions! So this new ID will be for the measure with every version change. It will be generated from client side when a new measure is created, for measures previously generated, the current measure Id can be its version ID. When there is version action, a new version ID will be generated, and that will be the measure Id. (this part will be implemented later).

I do have changes in madie-layout for new measure and madie-measure for updating measure in terms of version ID.

[the remark this answers is not in the document]
Comment (adongare) [reply src/main/java/gov/cms/madie/models/measure/Measure.java:105]: got it. thank you for the clarification!

**Review by adongare — APPROVED** [review]
